src/vs/workbench/browser/quickopen.ts

```diff
@@ -247,20 +248,38 @@ export class EditorQuickOpenEntry extends QuickOpenEntry implements IEditorQuick
 	}
 
 	public run(mode: Mode, context: IEntryRunContext): boolean {
-		if (mode === Mode.OPEN) {
+		const hideWidget = mode === Mode.OPEN;
+
+		let backgroundOpts;
+		if (mode === Mode.OPEN_IN_BACKGROUND) {
+			backgroundOpts = objects.assign({}, {
+				pinned: true,
+				preserveFocus: true
+			});
+		}
+
+		if (mode === Mode.OPEN || mode === Mode.OPEN_IN_BACKGROUND) {
 			let sideBySide = context.keymods.indexOf(KeyMod.CtrlCmd) >= 0;
+			let opts = this.getOptions();
+			if (backgroundOpts) {
+				opts = objects.assign(opts || {}, backgroundOpts);
```

Comment:
You cannot use `objects.assign` with the `opts` here because opts can be a real instance of `EditorOptions`, it might not just be an options bag. Maybe the better fix is to see if `this.getOptions()` could always return `IEditorOptions`.